cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/SessionManager.java

```diff
@@ -184,7 +199,12 @@ synchronized void expireOldSessions() {
         LOG.info("Expiring session {}.", session);
         iter.remove();
         session.invalidate();
-        info.lastFuture().cancel(true);
+        _sessionLifetimeTimer.update(System.nanoTime() - info.requestTime(), TimeUnit.NANOSECONDS);
+        if (info.lastFuture().isDone() && info.executionTime() > 0) {
+          _requestExecutionTimer.get(info.endPoint()).update(info.executionTime(), TimeUnit.NANOSECONDS);
+        } else {
```

Comment:
Isn't this a behavior change? Shouldn't `info.lastFuture().cancel(true);` be out of the else statement?

Reply:
If the `future.isDone()` returns True, then `future.cancel()` does not have any effect. I think it is safe to call `cancel()` only in else branch.